aptos-move/replay-benchmark/src/generator.rs

```diff
@@ -1,105 +1,166 @@
                         num_generated.fetch_add(1, Ordering::SeqCst) + 1,
                         num_blocks,
                         time
                     );
-                    block
+                    (inputs, diffs)
                 }
             });
             tasks.push(task);
         }
 
-        let mut blocks = Vec::with_capacity(tasks.len());
+        let mut all_inputs = Vec::with_capacity(tasks.len());
+        let mut all_diffs = Vec::with_capacity(tasks.len());
         for task in tasks {
-            blocks.push(task.await?);
+            let (inputs, diffs) = task.await?;
+            all_inputs.push(inputs);
+            all_diffs.push(diffs)
         }
 
-        Ok(blocks)
+        Ok((all_inputs, all_diffs))
     }
 
-    /// Generates a single [Block] for benchmarking.
-    fn generate_block(&self, begin: Version, txns: Vec<Transaction>) -> Block {
-        let workload = Workload::new(begin, txns);
-        let state_view = self.debugger.state_view_at_version(begin);
-        let state_override = self.override_config.get_state_override(&state_view);
-        Block::new(workload, &state_view, state_override)
-    }
+    /// Generates a pre-block for a single block of transactions.
+    ///
+    /// Transactions are first executed on top of an actual state. Then, transactions are executed
+    /// in top of an overridden state. The outputs of two executions are compared, and the diffs
+    /// for each transaction are returned together with the inputs.
+    ///
+    /// Note: transaction execution is sequential, so that multiple inputs can be constructed in
+    /// parallel.
+    fn generate_inputs_with_diffs(
+        &self,
+        txn_block: TransactionBlock,
+    ) -> (ReadSet, Vec<TransactionDiff>) {
+        // For later comparison of outputs, find fee payers for transactions.
+        let fee_payers = txn_block
+            .transactions
+            .iter()
+            .map(|txn| {
+                txn.try_as_signed_user_txn().map(|txn| {
+                    txn.authenticator_ref()
+                        .fee_payer_address()
+                        .unwrap_or_else(|| txn.sender())
+                })
+            })
+            .collect::<Vec<_>>();
 
-    /// Partitions a sequence of transactions into blocks.
-    fn partition(&self, txns: Vec<Transaction>) -> Vec<(Version, Vec<Transaction>)> {
-        let mut begin_versions_and_blocks = Vec::with_capacity(txns.len());
+        let state_view = self.debugger.state_view_at_version(txn_block.begin_version);
+        let state_override = self.override_config.get_state_override(&state_view);
+        let workload = Workload::from(txn_block);
 
-        let mut curr_begin = self.begin_version;
-        let mut curr_block = Vec::with_capacity(txns.len());
+        // First, we execute transactions without overrides. Flush warm VM cache to ensure read-set
+        // contains modules used to warming up the VM.
+        flush_warm_vm_cache();
```

Comment:
Does this flush V2 code cache or in other words, how is this affected by V1/V2 loader distinction?

Reply:
Only V1 code, because V1 uses warm VM cache and V2 does not. Will be gone in a month once V2 is enabled.